### DISCUSSION TITLE
Mixcloud support

### DISCUSSION THREAD

**Discussion by funkytone** [music-assistant/support#2229] [Feature requests and ideas]
Title says it all - I would LOVE to see support for Mixcloud!

**Comment by tradergene** [comment]
Love MixCloud, this would be a killer feature.

**Comment by FoxxMD** [comment]
Please!! I do most of my listening on mixcloud. Being able to seamlessly move between long mixes and tracks would be huge.

**Comment by nukleuz80** [comment]
I would love it too! 🔥

**Comment by denbeiren** [comment]
Yes please :-)

**Comment by DrDagless** [comment]
Would absolutely love this as well

**Comment by max246** [comment]
I think it wont be possible at the moment: The audio streams are not available through the Mixcloud API. There are two reasons for this.

**Reply by max246** [reply]
Actually never mind, you could do it using this:

# Stream the video being downloaded to stdout
$ yt-dlp -o - BaW_jenozKc

I am not sure what is the policy about implementing yt-dlp

**Reply by OzGav** [reply]
Downloading music is not allowed.

**Comment by nldroid** [comment]
As Max246 said. The API doesn't support streaming: https://www.mixcloud.com/developers/#audio-streams

Probably the only way to support streaming is with a hacky method.

This request can be closed i guess..

**Reply by adamcstephens** [reply]
https://github.com/danielvijge/lms_mixcloud uses the API for information, but executes yt-dlp for the stream. So a bit of a hack, but workable it seems.

**Reply by OzGav** [reply]
Thanks for the suggestion, but I have to flag a legal issue with this approach.
 
The LMS Mixcloud plugin relies on the third-party downloader service mixcloud-downloader.com, which violate Mixcloud's terms of service and copyright agreements. 
 
Mixcloud's position is clear: 
* Mixcloud explicitly prohibits downloading their content for licensing reasons 
* They only have licenses for streaming, not downloading 
* Their terms specifically forbid circumventing these restrictions 

mixcloud-downloader.com which your suggested plugin uses appears to no longer exist but similar sites circumvent Mixcloud's streaming-only licensing by extracting download URLs that Mixcloud never intended to be downloadable.

**Comment by cieska** [comment]
Did maybe anyone note that anything has changed and mixcloud can be now be implemented for streaming? Sonos has it in provider list.

**Reply by cieska** [reply]
Seems there are some good sounds coming out https://help.mixcloud.com/hc/en-us/articles/360004185139-Using-the-Mixcloud-API

**Reply by adamcstephens** [reply]
They still say Audio Streams are not available. https://www.mixcloud.com/developers/#audio-streams

**Comment by OzGav** [comment]
Just looked into this and unfortunately, integrating Mixcloud natively into Music Assistant isn't possible due to API streaming restrictions. 

While Mixcloud does have a public developer API, they explicitly block third-party access to raw audio streams for licensing and copyright reasons. Their API only allows external apps to fetch metadata (like show names or tracklists) or load their embedded web widget—it does not provide the direct audio links that Music Assistant needs to stream music to your speakers.

Because bypassing this restriction violates their terms of service, we cannot build a native plugin. As an alternative, we recommend checking out SoundCloud, which is supported natively by Music Assistant and offers a similar platform for DJ sets and independent creators
